deps/wazuh_testing/wazuh_testing/logcollector.py

```diff
@@ -690,6 +691,37 @@ def callback_invalid_state_interval(interval):
     return monitoring.make_callback(pattern=msg, prefix=prefix, escape=True)
 
 
+def callback_logcollector_started():
+    """Check if logcollector started."""
+    return monitoring.make_callback(pattern='Started', prefix=prefix)
+
+
+def callback_log_bad_predicate():
+    """Check for the macOS ULS bad predicate message."""
+    return monitoring.make_callback(pattern="Execution error 'log:", prefix=prefix)
+
+
+def callback_macos_uls_log(expected_message):
+    """Callback function to wait for a macOS' ULS log, collected by logcollector."""
+    return monitoring.make_callback(pattern=expected_message, prefix=prefix, escape=False)
+
+
+def callback_logcollector_log_stream_log():
+    """Check for logcollector's macOS ULS module start message."""
+    return monitoring.make_callback(pattern='Monitoring macOS logs with:(.+?)log stream',
+                                    prefix=prefix, escape=False)
+
+
+def callback_file_status_macos_key(line):
+    """Check for 'macos' key."""
+    return monitoring.make_callback(pattern='"macos"', prefix='')
```

Comment:
This must be a callback since the file_status.json file changes along time, and we need to give `logcollector` a time window for this key to be added, we can not just open it and check once if the key is there or not.